api/lib/rest-to-tx-converter.js

```diff
@@ -70,24 +96,20 @@ RestToTxConverter.prototype.convert = function(payment, callback) {
     if (payment.destination_tag) {
       transaction.destinationTag(parseInt(payment.destination_tag, 10));
     }
+
     // SendMax
-    if (payment.source_amount) {
-      // Only set send max if source and destination currencies are different
-      if (!(payment.source_amount.currency
-            === payment.destination_amount.currency
-            && payment.source_amount.issuer
-            === payment.destination_amount.issuer)) {
-        var max_value = bignum(payment.source_amount.value)
-                        .plus(payment.source_slippage || 0).toString();
-        if (payment.source_amount.currency === 'XRP') {
-          transaction.sendMax(utils.xrpToDrops(max_value));
-        } else {
-          transaction.sendMax({
-            value: max_value,
-            currency: payment.source_amount.currency,
-            issuer: payment.source_amount.issuer
-          });
-        }
+    if (sendMaxRequired()) {
```

Comment:
:+1: on the extract method refactor here. Maybe a better name would be `isSendMaxRequired`.